pydantic/deprecated/parse.py

```diff
@@ -46,11 +46,11 @@ def load_str_bytes(
     if proto == Protocol.json:
         if isinstance(b, bytes):
             b = b.decode(encoding)
-        return json_loads(b)
+        return json_loads(b)  # type: ignore
     elif proto == Protocol.pickle:
         if not allow_pickle:
             raise RuntimeError('Trying to decode with pickle with allow_pickle=False')
-        bb = b if isinstance(b, bytes) else b.encode()
+        bb = b if isinstance(b, bytes) else b.encode()  # type: ignore
```

Comment:
Another solution: set ` disableBytesTypePromotions` ([docs](https://microsoft.github.io/pyright/#/configuration?id=type-check-diagnostics-settings))

Reply:
Hmm, @dmontagu what do you think?

Reply:
I think the #type: ignores are fine given it's deprecated code; we don't need to change the setting now. I am not necessarily opposed to changing the setting if it makes sense in the future though. I haven't looked, just saying for deprecated code I think it's fine to add # type: ignores without thinking about it.

Reply:
Probably fine to keep the type ignores indeed as it is deprecated. According to pyright's maintainer `disableBytesTypePromotions` is probably going to become the default as [PEP 688](https://peps.python.org/pep-0688/) comes into play.